ui/plugins/eu.esdihumboldt.hale.ui.common/src/eu/esdihumboldt/hale/ui/common/definition/viewer/OptionalPropertiesFilter.java

```diff
@@ -77,15 +86,22 @@ private boolean areChildrenOptional(EntityDefinition entityDef) {
 
 			Definition<?> def = child.getDefinition();
 
-			if (def instanceof PropertyDefinition) {
-				Cardinality cardinality = ((PropertyDefinition) def)
+			if (def instanceof GroupPropertyDefinition) {
+				Cardinality cardinality = ((GroupPropertyDefinition) def)
 						.getConstraint(Cardinality.class);
-				if (cardinality.getMinOccurs() != 0)
+				if (cardinality.getMinOccurs() != 0 && !areChildrenOptional(entityDef))
 					return false;
 			}
+			else if (def instanceof PropertyDefinition) {
+				Cardinality cardinality = ((PropertyDefinition) def)
+						.getConstraint(Cardinality.class);
 
-			if (!areChildrenOptional(child))
-				return false;
+				if (cardinality.getMinOccurs() != 0 && ((PropertyDefinition) def)
+						.getConstraint(NillableFlag.class).isEnabled()) {
```

Comment:
But if minOccurs != 0 and the NillableFlag is NOT enabled you should return false here, without checking the children.